test/suites/integration/tags_test.go

```diff
@@ -82,7 +83,42 @@ var _ = Describe("Tags", func() {
 			instance := env.GetInstance(pod.Spec.NodeName)
 			Expect(instance.SpotInstanceRequestId).ToNot(BeNil())
 			spotInstanceRequest := env.GetSpotInstanceRequest(instance.SpotInstanceRequestId)
-			Expect(tagMap(spotInstanceRequest.Tags)).To(HaveKeyWithValue("TestTag", "TestVal"))
+			Expect(spotInstanceRequest.Tags).To(ContainElement(&ec2.Tag{Key: lo.ToPtr("TestTag"), Value: lo.ToPtr("TestVal")}))
+		})
+		It("should tag managed instance profiles", func() {
+			nodeClass.Spec.Tags["TestTag"] = "TestVal"
+			env.ExpectCreated(nodeClass)
+
+			profile := env.EventuallyExpectInstanceProfileExists(env.GetInstanceProfileName(nodeClass))
+			Expect(profile.Tags).To(ContainElements(
+				&iam.Tag{Key: lo.ToPtr(fmt.Sprintf("kubernetes.io/cluster/%s", env.ClusterName)), Value: lo.ToPtr("owned")},
+				&iam.Tag{Key: lo.ToPtr(v1.LabelNodeClass), Value: lo.ToPtr(nodeClass.Name)},
+				&iam.Tag{Key: lo.ToPtr(v1.EKSClusterNameTagKey), Value: lo.ToPtr(env.ClusterName)},
+			))
+		})
+		It("should tag managed instance profiles with the eks:eks-cluster-name tag key after restart", func() {
+			env.ExpectCreated(nodeClass)
+			env.EventuallyExpectInstanceProfileExists(env.GetInstanceProfileName(nodeClass))
+
+			_, err := env.IAMAPI.UntagInstanceProfile(&iam.UntagInstanceProfileInput{
```

Comment:
Is this another permission in our cloud formation that we need to give explicitly to our e2es?

Reply:
Yep, already added :)